server/cli/src/report_utils.rs

```diff
@@ -0,0 +1,262 @@
+    }
+    Ok(())
+}
+
+fn process_report(
+    reports_data: &mut ReportsData,
+    path: &PathBuf,
+    con: &StorageConnection,
+) -> Result<(), Error> {
+    // install esbuild depedencies
+
+    if let Err(e) = run_yarn_install(&path) {
+        eprintln!("Failed to run yarn install in {}: {}", path.display(), e);
+    }
+    // read manifest file
+    let manifest_file = fs::File::open(path.join("report-manifest.json"))
+        .map_err(|e| Error::CannotOpenManifestFile(path.clone(), e))?;
+
+    let manifest: Manifest = serde_json::from_reader(manifest_file)
+        .map_err(|e| Error::CannotReadManifestFile(path.clone(), e))?;
+    let code = manifest.code;
+
+    let version = manifest.version;
+    let id_version = str::replace(&version, ".", "_");
+
+    let context = manifest.context;
+    let report_name = manifest.name;
+    let is_custom = manifest.is_custom;
+    let id = format!("{code}_{id_version}_{is_custom}");
+    let sub_context: Option<String> = manifest.sub_context;
+    let arguments_path = manifest
+        .arguments
+        .clone()
+        .and_then(|a| a.schema)
+        .and_then(|schema| Some(path.join(schema)));
+    let arguments_ui_path = manifest
+        .arguments
+        .and_then(|a| a.ui)
+        .and_then(|ui| Some(path.join(ui)));
+    let graphql_query = manifest.queries.clone().and_then(|q| q.gql);
+    let sql_queries = manifest.queries.clone().and_then(|q| q.sql);
+    let convert_data = manifest.convert_data.and_then(|cd| Some(path.join(cd)));
+    let custom_wasm_function = manifest.custom_wasm_function;
+    let query_default = manifest.query_default;
+
+    let args = BuildArgs {
+        dir: path.join("src"),
+        output: Some(path.join("generated").join("built_report.json")),
+        template: "template.html".to_string(),
+        header: manifest.header,
+        footer: manifest.footer,
+        query_gql: graphql_query,
+        query_default: query_default,
+        query_sql: sql_queries,
+        convert_data,
+        custom_wasm_function,
+    };
+
+    let report_definition =
+        build_report_definition(&args).map_err(|e| Error::FailedToBuildReport(path.clone(), e))?;
+
+    let filter = ReportFilter::new().id(EqualFilter::equal_to(&id));
+    let existing_report = ReportRepository::new(&con)
+        .query_by_filter(filter)
+        .map_err(|e| Error::RepositoryError(path.clone(), e))?
+        .pop();
+
+    let argument_schema_id =
+        existing_report.and_then(|r| r.argument_schema.as_ref().map(|r| r.id.clone()));
+
+    let form_schema_json = match (arguments_path, arguments_ui_path) {
+        (Some(_), None) | (None, Some(_)) => {
+            return Err(Error::FailedToGenerateArgumentSchema(path.to_path_buf()))
+        }
+        (Some(arguments_path), Some(arguments_ui_path)) => Some(
+            schema_from_row(FormSchemaRow {
+                id: argument_schema_id.unwrap_or(format!("for_report_{}", id)),
```

Comment:
I think format!("for_report_{}") is enough for all cases. Getting some id from local database doesn't make sense to me, reports are not database specific. Also I don't think that report making should rely on any database operations, please remove con. (It's not a new change in this PR, i understand, but would be great to get rid of it for now )